uamqp/_connection.py

```diff
@@ -470,6 +470,8 @@ def _process_outgoing_frame(self, channel, frame):
         
         :raises ValueError: If the connection is not open or not in a valid state.
         """
+        if self._network_trace:
+            _LOGGER.info("-> %r", frame, extra=self._network_trace_params)
```

Comment:
right, it would duplicate the logging.

I checked in uamqp-c that frames are logged in the connection layer. I personally like the idea of a centralized place (in connection) to log incoming/outgoing frames.

probably we could remove the log of outgoing/incoming frames in other modules like sender?

Reply:
I would also prefer to centralize - I know there was a reason I did it this way.... maybe it was to support the level-specific trace params.... or maybe it was because the frame was already encoded at the connection level.... either way, we can open an issue to figure it out at some point.